unit_testing/tests/userDataHandler.test.js

```diff
@@ -0,0 +1,188 @@
+const UserDataHandler = require('../src/data_handlers/user_data_handler.js')
+const nock = require('nock')
+const userData = require('../data/users.json')
+
+const testUserData = userData[1]
+let userDataHandler
+
+function setupMock () {
+  if (process.env.MOCK) {
+    nock(global.SERVER_URL)
+      .get('/users')
+      .reply(200, userData)
+  }
+}
+
+describe('when users were not loaded', () => {
+  beforeEach(async () => {
+    setupMock()
+    userDataHandler = new UserDataHandler()
+  })
+
+  afterEach(() => {
+    nock.cleanAll()
+  })
+
+  test('loadUsers should load users', async () => {
+    await userDataHandler.loadUsers()
+    expect(userDataHandler.users.length).toBeGreaterThan(0)
+    expect(userDataHandler.users[0]).toHaveProperty('name')
+  })
+
+  test('getUserEmailsList should throw error when no users', () => {
+    expect(() => userDataHandler.getUserEmailsList())
+      .toThrow('No users loaded!')
+  })
+
+  test('getNumberOfUsers should return 0 number of users', async () => {
+    expect(userDataHandler.getNumberOfUsers()).toBe(0)
+  })
+
+  test('should throw error when no users', async () => {
+    const searchParams = { name: 'Yauhen' }
+    expect(() => userDataHandler.findUsers(searchParams))
+      .toThrow('No users loaded!')
+  })
+})
+
+describe('when users loading failed', () => {
+  beforeAll(async () => {
+    nock.cleanAll()
+    nock(global.SERVER_URL)
+      .get('/users')
+      .replyWithError('Connection refused')
+    userDataHandler = new UserDataHandler()
+  })
+
+  afterAll(() => {
+    nock.cleanAll()
+  })
```

Comment:
just to be safe in case we want to add more tests in this context later
```suggestion
  beforeEach(async () => {
    nock(global.SERVER_URL)
      .get('/users')
      .replyWithError('Connection refused')
    userDataHandler = new UserDataHandler()
  })

  afterEach(() => {
    nock.cleanAll()
  })
```

Reply:
Agreed. Commited.